api/types/app.go

```diff
@@ -537,3 +539,27 @@ func (a *AppIdentityCenter) GetPermissionSets() []*IdentityCenterPermissionSet {
 	}
 	return a.PermissionSets
 }
+
+// PortRanges is a list of port ranges.
+type PortRanges []*PortRange
+
+// Contains checks if targetPort is within any of the port ranges.
+func (p PortRanges) Contains(targetPort int) bool {
+	return slices.ContainsFunc(p, func(portRange *PortRange) bool {
+		return netutils.IsPortInRange(int(portRange.Port), int(portRange.EndPort), targetPort)
+	})
+}
+
+// Len returns the slice length.
+func (p PortRanges) Len() int { return len(p) }
+
+// Less compares port ranges by port and end port.
+func (p PortRanges) Less(i, j int) bool {
+	return cmp.Or(
+		cmp.Compare(p[i].Port, p[j].Port),
+		cmp.Compare(p[i].EndPort, p[j].EndPort),
+	) == -1
+}
+
+// Swap swaps two port ranges.
+func (p PortRanges) Swap(i, j int) { p[i], p[j] = p[j], p[i] }
```

Comment:
imo slices.SortFunc is usually a better choice than sort.Sort these days, then you could just implement a comparison instead of the full sort.Interface, this is a fairly minor difference but if we're not even using this code I'd delete it, maybe implement a Compare method if you want to sort later with slices.SortFunc

Reply:
I just removed it completely then.